### PR TITLE
Secure Boot & Virtual TPM

### PR DESCRIPTION
**Pull request by stejskalleos**
Add support for Secure Boot and TPM
Inspired by https://github.com/ananace/foreman_vmware_advanced

### CODE REVIEW

**Comment by chris1984** [comment]
@stejskalleos is this good to review?

**Comment by stejskalleos** [comment]
Yes, but I cannot say that 100% it's working as my provisioning was kinda funky when I was testing it.

**Review by chris1984 — COMMENTED** [review]
Seeing this when I try to provision:

```bash
Failed to create a compute VMware8 (VMware) instance [irene-pendl.satellite.lab.eng.rdu2.redhat.com](http://irene-pendl.satellite.lab.eng.rdu2.redhat.com/): RuntimeFault: RuntimeFault.summary
```

File: lib/fog/vsphere/requests/compute/create_vm.rb [hunk 1]
```diff
@@ -166,6 +169,10 @@ def boot_options(attributes, vm_cfg)
             options[:bootRetryDelay]   = attributes[:boot_retry]
           end
 
+          if attributes[:secure_boot]
+            options[:efiSecureBootEnabled] = true
+          end
```
Comment (nofaralfasi) [inline lib/fog/vsphere/requests/compute/create_vm.rb:174]: This won't work if `boot_order` or `boot_retry` are not set. See https://github.com/fog/fog-vsphere/blob/62fb4b9b9b4e1071cd7fd6632244aa0896297a5e/lib/fog/vsphere/requests/compute/create_vm.rb#L25

**Comment by nofaralfasi** [comment]
I don't have permission to push changes to this PR, so I created a new one: https://github.com/fog/fog-sphere/pull/305.

**Pull request closed without merging**